app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,74 @@
+class AuthorsController < ApplicationController
+  before_action :set_author, only: [:show, :edit, :update, :destroy]
+
+  # GET /authors
+  # GET /authors.json
+  def index
+    @authors = Author.all
+  end
+
+  # GET /authors/1
+  # GET /authors/1.json
+  def show
+  end
+
+  # GET /authors/new
+  def new
+    @author = Author.new
+  end
+
+  # GET /authors/1/edit
+  def edit
+  end
+
+  # POST /authors
+  # POST /authors.json
+  def create
+    @author = Author.new(author_params)
+
+    respond_to do |format|
+      if @author.save
+        format.html { redirect_to @author, notice: 'Author was successfully created.' }
+        format.json { render :show, status: :created, location: @author }
+      else
+        format.html { render :new }
+        format.json { render json: @author.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /authors/1
+  # PATCH/PUT /authors/1.json
+  def update
+    respond_to do |format|
+      if @author.update(author_params)
+        format.html { redirect_to @author, notice: 'Author was successfully updated.' }
+        format.json { render :show, status: :ok, location: @author }
+      else
+        format.html { render :edit }
+        format.json { render json: @author.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /authors/1
+  # DELETE /authors/1.json
+  def destroy
+    @author.destroy
+    respond_to do |format|
+      format.html { redirect_to authors_url, notice: 'Author was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
```

Comment:
having comments is a bad practice in ruby.
avoid intuitive comments like this one.
Most comments should be understood from reading the code and using proper naming conventions. 
If you feel the urge to convey an important message add it in the commit message.